### PR TITLE
User, Address, products related features

### PR DESCRIPTION
**Pull request by flutterindiadev**
- get user profile feature 
- get user favorite products
- Edit address
- search products feature
- userexists feature

### CODE REVIEW

File: core/data/lib/src/out/network_port.dart [hunk 1]
```diff
@@ -1,43 +1,65 @@
...
+  Future<Either<NetworkError, Logout>> logout(
+      {required LogoutRequest logoutRequest});
+
+  Future<Either<NetworkError, List<Product>>> getUserFavoriteProducts(
+      {required int userId});
+
+  Future<Either<NetworkError, UserProfile>> getUserProfile(
+      {required int userId});
+
+  Future<Either<NetworkError, UserProfile>> uodateUserProfile(
```
Comment (naikdp7) [inline core/data/lib/src/out/network_port.dart]: correct the typo

File: core/data/lib/src/repository/cart_repository_impl.dart [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
   final NetworkPort networkPort;
   CartRepositoryImpl(this.networkPort);
   @override
-  Future<Either<NetworkError, Cart>> addToCart (
-      {required AddtoCartRequest addtoCartRequest}) async{
-
-    return  await networkPort.addtoCart( addtoCartRequest: addtoCartRequest,);
+  Future<Either<NetworkError, Cart>> addToCart(
+      {required AddtoCartRequest addtoCartRequest}) async {
+    return await networkPort.addtoCart(
```
Comment (naikdp7) [inline core/data/lib/src/repository/cart_repository_impl.dart]: avoid unnecessary await

File: core/domain/lib/src/usecase/address/edit_address_usecase.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      required this.country,
+      required this.city,
+      required this.pincode,
+      required this.floor,
+      required this.building,
+      required this.apartment});
+
+  @override
+  Either<AppError, bool> verify() {
+    return Right(true);
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/address/edit_address_usecase.dart]: validations?

File: core/domain/lib/src/usecase/products/select_category_usecase.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+class SelectCategoryUsecaseParams extends Params {
+  String category;
+
+  SelectCategoryUsecaseParams(this.category);
+
+  @override
+  Either<AppError, bool> verify() {
+    if (category == '') {
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/products/select_category_usecase.dart]: use category.isEmpty

File: core/domain/lib/src/usecase/user/change_password_usecase.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  ChangePasswordUsecaseParams(
+      {super.reloading,
+      required this.userId,
+      required this.newPassword,
+      required this.token});
+
+  @override
+  Either<AppError, bool> verify() {
+    // TODO: implement verify
+    throw UnimplementedError();
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/user/change_password_usecase.dart]: validations?

File: core/domain/lib/src/usecase/user/update_user_profile_usecase.dart [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      required this.pincode,
+      required this.floor,
+      required this.building,
+      required this.apartment,
+      required this.orderHistory,
+      required this.favoriteProducts});
+
+  @override
+  Either<AppError, bool> verify() {
+    return Right(true);
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/user/update_user_profile_usecase.dart]: validations?

File: core/domain/lib/src/usecase/user/user_settings_usecase.dart [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+class UserSettingsUsecaseParams extends Params {
+  int userId;
+
+  UserSettingsUsecaseParams(this.userId);
+
+  @override
+  Either<AppError, bool> verify() {
+    if (userId == 0) {
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/user/user_settings_usecase.dart:24]: use common function for this validation in other usecases

**Review by naikdp7 — CHANGES_REQUESTED** [review]
Requested changes

File: core/data/lib/src/out/network_port.dart (hunk 1 above)
Comment (flutterindiadev) [reply core/data/lib/src/out/network_port.dart]: Changed

File: core/data/lib/src/repository/cart_repository_impl.dart (hunk 2 above)
Comment (flutterindiadev) [reply core/data/lib/src/repository/cart_repository_impl.dart]: Done

File: core/domain/lib/src/usecase/address/edit_address_usecase.dart (hunk 3 above)
Comment (flutterindiadev) [reply core/domain/lib/src/usecase/address/edit_address_usecase.dart]: Added

File: core/domain/lib/src/usecase/products/select_category_usecase.dart (hunk 4 above)
Comment (flutterindiadev) [reply core/domain/lib/src/usecase/products/select_category_usecase.dart]: Done

File: core/domain/lib/src/usecase/user/change_password_usecase.dart (hunk 5 above)
Comment (flutterindiadev) [reply core/domain/lib/src/usecase/user/change_password_usecase.dart]: Done

File: core/domain/lib/src/usecase/user/update_user_profile_usecase.dart (hunk 6 above)
Comment (flutterindiadev) [reply core/domain/lib/src/usecase/user/update_user_profile_usecase.dart]: Done

File: core/domain/lib/src/usecase/user/user_settings_usecase.dart (hunk 7 above)
Comment (flutterindiadev) [reply core/domain/lib/src/usecase/user/user_settings_usecase.dart:24]: Done

**Comment by flutterindiadev** [comment]
Fixed the issues

**Review by naikdp7 — APPROVED** [review]

**Pull request merged**